src/bit_stream_reader/es_reader.cpp

```diff
@@ -0,0 +1,1149 @@
+/*
+Copyright (c) 2023 - 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <string.h>
+#include "es_reader.h"
+#include "hevc_defines.h"
+#include "avc_defines.h"
+#include "av1_defines.h"
+#include "roc_video_parser.h"
+
+RocVideoESParser::RocVideoESParser(const char *input_file_path) {
+    p_stream_file_.open(input_file_path, std::ifstream::in | std::ifstream::binary);
+    if (!p_stream_file_) {
+        ERR("Failed to open the bitstream file.");
+    }
+    end_of_file_ = false;
+    end_of_stream_ = false;
+    bs_ring_ = static_cast<uint8_t*>(malloc(BS_RING_SIZE));
```

Comment:
It is recommended to use C++ `new` and `delete` for memory allocation and deallocation instead of using `malloc` and `free` functions.

Why is dynamic allocation needed for `bs_ring_` if its size is always `BS_RING_SIZE`? Could you use a fixed array for `bs_ring_` instead?

Reply:
Done.